interpreter/binary/decode.ml

```diff
@@ -1059,31 +1060,31 @@ let module_ s =
   require (header = magic) s 0 "magic header not detected";
   let version = u32 s in
   require (version = Encode.version) s 4 "unknown binary version";
-  iterate custom_section s;
+  let customs = iterate (custom_section Custom.(Before Type)) s in
```

Comment:
Nit: Analogous to the encoder, perhaps do a `let open Custom in ...` at the top of the function and save the repeated `Custom.`?